packages/sources/superstate/src/transport/nav.ts

```diff
@@ -0,0 +1,201 @@
+import { Transport, TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/nav'
+import { isWeekend } from 'date-fns'
+import {
+  AdapterRequest,
+  AdapterResponse,
+  makeLogger,
+} from '@chainlink/external-adapter-framework/util'
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { calculateHttpRequestKey } from '@chainlink/external-adapter-framework/cache'
+import schedule from 'node-schedule'
+import {
+  getPreviousNonWeekendDay,
+  getStartingAndEndingDates,
+  isBeforeTime,
+  isInTimeRange,
+} from './utils'
+
+const logger = makeLogger('Superstate')
+
+export interface ResponseSchema {
+  fund_id: number
+  net_asset_value_date: string
+  net_asset_value: string
+  assets_under_management: string
+  outstanding_shares: string
+  net_income_expenses: string
+}
+
+const RETRY_INTERVAL_MINUTES = 1
+const TZ = 'America/New_York'
+
+// Custom transport implementation that takes incoming requests, adds them into a SET, and makes requests to DP
+// on a specific time every day, after receiving a signal from scheduler.
+export class NavTransport implements Transport<BaseEndpointTypes> {
+  name!: string
+  responseCache!: ResponseCache<BaseEndpointTypes>
+  requester!: Requester
+  settings!: BaseEndpointTypes['Settings']
+  endpointName!: string
+  fundIdsSet!: Set<number>
+
+  async initialize(
+    dependencies: TransportDependencies<BaseEndpointTypes>,
+    settings: BaseEndpointTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    this.responseCache = dependencies.responseCache
+    this.name = transportName
+    this.requester = dependencies.requester
+    this.settings = settings
+    this.endpointName = endpointName
+    this.fundIdsSet = new Set()
+    this.runScheduler()
+  }
+
+  // registerRequest is invoked on every valid request to EA
+  // Adds fundId in the request to a SET
+  async registerRequest(req: AdapterRequest<typeof inputParameters.validated>) {
+    const { fundId } = req.requestContext.data
+    if (!this.fundIdsSet.has(fundId)) {
+      this.fundIdsSet.add(fundId)
+      logger.info(`Added new fund id - ${fundId}`)
+    }
+  }
+
+  // foregroundExecute is executed when there is a new request/fundId that is not in the cache
+  async foregroundExecute(
+    req: AdapterRequest<typeof inputParameters.validated>,
+  ): Promise<AdapterResponse<BaseEndpointTypes['Response']> | void> {
+    const { fundId } = req.requestContext.data
+    return this.execute(fundId)
+  }
+
+  // Runs 'execute' function every day at 9:09 AM ET (if fundIdsSet is not empty)
+  runScheduler() {
+    const rule = new schedule.RecurrenceRule()
+    rule.hour = 9
+    rule.minute = 9
+    rule.tz = TZ
```

Comment:
Does this library automatically figure out whether this should be Eastern Daylight Time or Eastern Standard Time?

Reply:
[Seems](https://github.com/harrisiirak/cron-parser/issues/111) that it supports DSTs